### DISCUSSION TITLE
Detect if app was installed through TestFlight

### DISCUSSION THREAD

**Discussion by RocMark** [expo/expo#19679] [Archived: Questions]
Is there a way to detect if app was installed through Testfight built in Expo?

There is some solution i found but not sure. how can i make it work with Expo
https://github.com/react-native-device-info/react-native-device-info/issues/228#issuecomment-420703355
https://stackoverflow.com/questions/18282326/how-can-i-detect-if-the-currently-running-app-was-installed-from-the-app-store/18619270

- native module can get isTestflight but need react-native link, but I don't want  to eject Expo
https://github.com/communitybuilders/react-native-test-flight
https://stackoverflow.com/questions/44977693/react-native-link-using-expo

**Comment by StCyr** [comment]
no much reaction here... And I'm wondering too

**Comment by jeremybarbet** [comment]
Never too late, but had the same issue on some of my projects. I wrote a simple expo module for it https://www.npmjs.com/package/expo-testflight https://github.com/jeremybarbet/expo-testflight

**Reply by felipeeleon** [reply]
appStoreReceiptURL is deprecated https://developer.apple.com/documentation/foundation/bundle/appstorereceipturl

**Comment by BenStirrup** [comment]
@jonsamp 
Could @jeremybarbet 's module / implementation make its way into the official `expo` API ?
The implementation is short but useful, therefore it's worth integrating into expo if you believe it is stable enough.

**Comment by tttocklll** [comment]
I built a package inspired by the store_checker library, which is the go-to solution in Flutter.

It handles iOS TestFlight detection properly, but it also covers diverse installation sources on Android (like detecting if the app was sideloaded or installed via a specific store).

You can check it out here:

https://github.com/tttocklll/store_checker
https://www.npmjs.com/package/expo-store-checker

Stars and feedback are welcome! ⭐️
